docs/docs/how_to.md

```diff
@@ -0,0 +1,227 @@
+directory prefixed with `rcj_soccer_player`. Each robot is programmed the same way,
+meaning it rotates to face the ball and goes towards it afterwards.
+
+We decided to make the code reusable and put some common methods into
+`RCJSoccerRobot` class inside `rcj_soccer_robot.py` file.
+
+Let's put together a simple program on how to program the robot.
+
+```python
+from controller import Robot
+import struct
+
+TIME_STEP = 64
+ROBOT_NAMES = ["B1", "B2", "B3", "Y1", "Y2", "Y3"]
+N_ROBOTS = len(ROBOT_NAMES)
+
+
+class MyRobot:
+    def __init__(self):
+        self.robot = Robot()
+        self.name = self.robot.getName()
+
+        self.receiver = self.robot.getDevice("receiver")
+        self.receiver.enable(TIME_STEP)
+
+        self.left_motor = self.robot.getDevice("left wheel motor")
+        self.right_motor = self.robot.getDevice("right wheel motor")
+
+        self.left_motor.setPosition(float('+inf'))
+        self.right_motor.setPosition(float('+inf'))
+
+        self.left_motor.setVelocity(0.0)
+        self.right_motor.setVelocity(0.0)
+    
+    def get_new_data(self):
+        packet = self.receiver.getData()
+        self.receiver.nextPacket()
+
+        struct_fmt = 'ddd' * 6 + 'dd'
+
+        unpacked = struct.unpack(struct_fmt, packet)
+
+        data = {}
+        for i, r in enumerate(ROBOT_NAMES):
+            data[r] = {
+                "x": unpacked[3 * i],
+                "y": unpacked[3 * i + 1],
+                "orientation": unpacked[3 * i + 2]
+            }
+        data["ball"] = {
+            "x": unpacked[3 * N_ROBOTS],
+            "y": unpacked[3 * N_ROBOTS + 1]
+        }
+        return data
+
+    def run(self):
+        while self.robot.step(TIME_STEP) != -1:
+            if self.receiver.getQueueLength() > 0:
+                data = self.get_new_data()
+
+                # Get the position of our robot
+                robot_pos = data[self.name]
+                # Get the position of the ball
+                ball_pos = data['ball']
+
+                self.left_motor.setVelocity(1)
+                self.right_motor.setVelocity(-1)
+
+
+my_robot = MyRobot()
+my_robot.run()
+```
+
+Let's explain the code in detail:
+
+```python
+from controller import Robot
+```
+
+The Robot class is required to be imported because this is the only way we are able
+to controll the robot. The Robot class is shipped together with webots.
+
+```python
+import struct
+```
+
+This library is a built-in python library, which is required to unpack the data sent by
+the supervisor.
+
+```python
+TIME_STEP = 64
+ROBOT_NAMES = ["B1", "B2", "B3", "Y1", "Y2", "Y3"]
+N_ROBOTS = len(ROBOT_NAMES)
+```
+
+We define some constants here, whose values we will be used later.
+
+```python
+class MyRobot:
+```
+
+You can wrap the program into the class as we did. The benefit of OOP
+(object oriented programmin) is that you can later inherit from common class used
```

Comment:
```suggestion
([Object Oriented Programming](https://realpython.com/python3-object-oriented-programming/)) is that you can later reuse the same common class
```

Reply:
@Adman I don't really like the cheap advertisement but it seems like a pretty nice link in this case